client.go

```diff
@@ -152,10 +152,16 @@ func (cli *Client) Dial(network, address string) (Conn, error) {
 	}
 
 	if strings.HasPrefix(network, "tcp") {
-		if cli.opts.TCPNoDelay == TCPNoDelay {
+		switch cli.opts.TCPNoDelay {
+		case TCPDelay:
```

Comment:
I think we only need to check if `cli.opts.TCPNoDelay == TCPDelay` since `net` has already set TCPNoDelay to true, so remove this `switch` and just modify the condition to `cli.opts.TCPNoDelay == TCPDelay`, then set TCPNoDelay to false.